azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/data_structures/method_request.py

```diff
@@ -0,0 +1,34 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains a class representing a request to invoke a direct method.
+"""
+
+
+class MethodRequest(object):
+    """Represents a request to invoke a direct method.
+
+    :ivar str name: The name of the method to be invoked
+    :ivar payload: The payload being sent with the request.
+    """
+
+    def __init__(self, request_id, name, payload):
+        """Initializer for a MethodRequest.
+
+        :param str request_id: The request id of the request.
+        :param str name: The name of the method to be invoked
+        :param payload: The payload being sent with the request.
+        """
+        self._request_id = request_id
```

Comment:
This probably needs to be made public, but I'll make that change with the transport implementation when it becomes necessary.